Ghidra/Features/Decompiler/src/decompile/cpp/pcodecompile.cc

```diff
@@ -623,7 +623,17 @@ vector<OpTpl *> *PcodeCompile::assignBitRange(VarnodeTpl *vn,uint4 bitoffset,uin
   bool shiftneeded = (bitoffset != 0);
   bool zextneeded = true;
   uintb mask = (uintb)2;
-  mask = ~(((mask<<(numbits-1))-1) << bitoffset);
+  int4 masknumbits = sizeof(mask) * 8;
+  if (numbits - 1 < masknumbits)
```

Comment:
It's invalid to left shift more bits (or an equal number of bits) than the width of the mask itself. To make this explicit, we should add a check and if `numbits` is larger, we expect it to get zero'd out.

Reply:
looks like this could get reduced to avoid all the extra assignments
```C
uintb mask = 0;
const int4 masknumbits = sizeof(mask) * 8;
if (numbits - 1 < masknumbits && bitoffset < masknumbits)
    mask = ~((((uintb)2<<(numbits-1))-1) << bitoffset);
```

Reply:
[23bd3d5](https://github.com/NationalSecurityAgency/ghidra/pull/3967/commits/23bd3d5672e81c6288f0585b7271a4f2366494eb)